src/posix/client.cpp

```diff
@@ -161,9 +161,9 @@ static bool ReconnectSession(void)
     return ok;
 }
 
-static bool IsSeparator(char aChar)
+static bool NeedEscape(char aChar)
```

Comment:
We defined `IsEscapable` in openthread: `utils/parse_cmdline.cpp`. 
nit: Can we use `IsEscapable` directly here?